controllers/vspheremachine_controller.go

```diff
@@ -388,12 +389,14 @@ func (r *machineReconciler) clusterToVSphereMachines(ctx context.Context, a clie
 }
 
 func (r *machineReconciler) fetchCAPICluster(ctx context.Context, machine *clusterv1.Machine, vsphereMachine metav1.Object) *clusterv1.Cluster {
-	log := ctrl.LoggerFrom(ctx)
-	cluster, err := clusterutilv1.GetClusterFromMetadata(r, r.Client, machine.ObjectMeta)
+	log := ctrl.LoggerFrom(ctx).WithValues("Machine", klog.KObj(machine), "VSphereMachine", klog.KRef(vsphereMachine.GetNamespace(), vsphereMachine.GetName()))
+	ctx = ctrl.LoggerInto(ctx, log)
+	cluster, err := clusterutilv1.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
 	if err != nil {
 		log.Info("Machine is missing cluster label or cluster does not exist")
 		return nil
 	}
+	log = log.WithValues("Cluster", klog.KObj(cluster))
 	if annotations.IsPaused(cluster, vsphereMachine) {
 		log.V(4).Info("VSphereMachine %s/%s linked to a cluster that is paused", vsphereMachine.GetNamespace(), vsphereMachine.GetName())
```

Comment:
VSphereMachine info is already in the logger as kv pair. Let's improve this log line:


```suggestion
		log.V(4).Info("VSphereMachine is linked to a cluster that is paused")
```